bu_passport/lib/pages/event_page.dart

```diff
@@ -48,64 +119,102 @@ class _EventPageState extends State<EventPage> {
+                    fontWeight: FontWeight.bold,
                   ),
-                  // Add more event details as needed
-                ],
-              ),
+                ),
+                SizedBox(height: sizedBoxHeight),
+                Text(
+                  'Location: ${widget.event.eventLocation}',
+                  style: TextStyle(fontSize: 16),
+                ),
+                SizedBox(height: sizedBoxHeight),
+                Text(
+                  'Start Time: ${DateFormat('h:mm a, EEEE, MMMM d, y').format(widget.event.eventStartTime)}',
+                  style: TextStyle(fontSize: 16),
+                ),
+                SizedBox(height: sizedBoxHeight),
+                Text(
+                  'End Time: ${DateFormat('h:mm a, EEEE, MMMM d, y').format(widget.event.eventEndTime)}',
+                  style: TextStyle(fontSize: 16),
+                ),
+                SizedBox(height: sizedBoxHeight),
+                Text(
+                  'Description: ${widget.event.eventDescription}',
+                  style: TextStyle(fontSize: 16),
+                ),
+                // Add more event details as needed
+              ],
             ),
           ),
           Padding(
             padding: EdgeInsets.all(edgeInsets),
-            child: ElevatedButton(
-              onPressed: () async {
-                String userUID = FirebaseAuth.instance.currentUser?.uid ?? "";
-                String eventId = widget.event.eventId;
-                bool isRegistered =
-                    await FirebaseService.isUserRegisteredForEvent(
-                        userUID, eventId);
-                if (isRegistered) {
-                  FirebaseService.unregisterFromEvent(userUID, eventId);
-                } else {
-                  FirebaseService.registerForEvent(userUID, eventId);
-                }
-                setState(() {
-                  _isRegistered = !_isRegistered; // Toggle registration status
-                });
-              },
-              child: Text(_isRegistered ? 'Unregister' : 'Register'),
+            child: Column(
+              children: [
+                ElevatedButton(
+                  onPressed: (!_isRegistered ||
+                          !isEventToday(widget.event.eventStartTime) ||
+                          _isCheckedIn)
+                      ? null
+                      : () async {
+                          // Your check-in logic here. On successful check-in, update the _isCheckedIn state.
+                          bool success = await checkIn();
+                          if (success) {
+                            ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+                                content: Text("Checked in successfully!")));
+                          } else {
+                            ScaffoldMessenger.of(context).showSnackBar(
+                                SnackBar(content: Text("Unable to check in: location too far!")));
```

Comment:
- Is "location too far" the only false case here, or there might be other possible false cases, such as "Location permission not granted"?